collect_app/src/main/java/org/odk/collect/android/activities/FormHierarchyActivity.java

```diff
@@ -145,6 +145,11 @@ public void onCreate(Bundle savedInstanceState) {
         formEntryViewModel = new ViewModelProvider(this, formEntryViewModelFactory).get(FormEntryViewModel.class);
 
         FormController formController = formEntryViewModel.getFormController();
+        if (formController == null) {
+            finish();
```

Comment:
This is how the previous version of Collect behaved, but there was no test, and we ended up missing the behaviour change in code review.